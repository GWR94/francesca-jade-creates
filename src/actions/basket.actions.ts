import {
  ADD_ITEM_TO_BASKET,
  CLEAR_BASKET,
  REMOVE_ITEM_FROM_BASKET,
  AddItemAction,
  ClearBasketAction,
  RemoveItemAction,
} from "../interfaces/basket.redux.i";
import { BasketItemProps } from "../pages/payment/interfaces/Basket.i";

export const addToBasket = (item: BasketItemProps): AddItemAction => ({
  type: ADD_ITEM_TO_BASKET,
  item,
});

export const clearBasket = (): ClearBasketAction => ({
  type: CLEAR_BASKET,
});

export const removeFromBasket = (itemID: string): RemoveItemAction => ({
  type: REMOVE_ITEM_FROM_BASKET,
  itemID,
});
