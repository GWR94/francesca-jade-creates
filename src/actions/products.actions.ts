import { Dispatch } from "redux";
import { API } from "aws-amplify";
import { listProducts } from "../graphql/queries";
import { ProductProps } from "../pages/accounts/interfaces/Product.i";
import {
  FETCH_PRODUCTS_SUCCESS,
  FETCH_PRODUCTS_FAILURE,
  SET_FILTERS,
  FetchProductsFailureAction,
  FetchProductsSuccessAction,
  SetFiltersAction,
  FilterActionProps,
} from "../interfaces/products.redux.i";

export const fetchProductsSuccess = (
  products: ProductProps[],
): FetchProductsSuccessAction => ({
  type: FETCH_PRODUCTS_SUCCESS,
  products,
});

export const fetchProductsFailure = (): FetchProductsFailureAction => ({
  type: FETCH_PRODUCTS_FAILURE,
});

export const setSearchFilters = (filters: FilterActionProps): SetFiltersAction => ({
  type: SET_FILTERS,
  filters,
});

export const getProducts = (type?: "Cake" | "Creates") => {
  return async (
    dispatch: Dispatch,
  ): Promise<FetchProductsSuccessAction | FetchProductsFailureAction> => {
    try {
      const { data } = await API.graphql({
        query: listProducts,
        variables: {
          filter: type && {
            type: {
              eq: type,
            },
          },
          limit: 100,
        },
        // @ts-ignore
        authMode: "API_KEY",
      });
      return dispatch(fetchProductsSuccess(data.listProducts.items));
    } catch (error) {
      return dispatch(fetchProductsFailure());
    }
  };
};
