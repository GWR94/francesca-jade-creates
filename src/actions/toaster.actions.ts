export const setSuccessToaster = (message) => ({
  type: SET_SUCCESS_TOASTER,
  message,
});

export const setErrorToaster = (message) => ({
  type: SET_ERROR_TOASTER,
  message,
});
