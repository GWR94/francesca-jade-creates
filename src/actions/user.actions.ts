import {
  SetUserAction,
  SET_USER,
  ClearUserAction,
  CLEAR_USER,
} from "../interfaces/user.redux.i";

export const setUser = (id: string): SetUserAction => ({
  type: SET_USER,
  id,
});

export const clearUser = (): ClearUserAction => ({
  type: CLEAR_USER,
});
