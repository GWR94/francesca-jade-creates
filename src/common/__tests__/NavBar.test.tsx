import React from "react";
import { shallow } from "enzyme";
import { Popover, Menu } from "@blueprintjs/core";
import configureStore from "redux-mock-store";
import ConnectedNavBar, { NavBar } from "../NavBar";
import * as actions from "../../actions/basket.actions";
import * as types from "../../interfaces/basket.redux.i";
import reducer from "../../reducers/basket.reducer";

describe("<NavBar />", () => {
  // mock logout and setAccounts functions to test they are being called
  const logout = jest.fn();
  const setAccounts = jest.fn();

  const initialState = {
    basket: {
      items: [],
      cost: 0,
    },
    user: {
      id: null,
    },
  };

  const firstItem = {
    id: "123",
    title: "first",
    description: "test desc",
    price: 6,
    shippingCost: 4,
    image: null,
    type: "Cake",
  };

  const secondItem = {
    id: "321",
    title: "second",
    description: "test desc",
    price: 6,
    shippingCost: 4,
    image: null,
    type: "Cake",
  };

  describe("test cases for UNCONNECTED component's rendering and snapshots", () => {
    it("should render correctly when user IS admin", () => {
      const wrapper = shallow(
        <NavBar
          signOut={logout}
          setAccountsTab={setAccounts}
          // @ts-ignore
          user={{
            username: "test",
          }}
          userAttributes={{
            sub: "123",
          }}
          admin
          items={[]}
          cost={0}
        />,
      );
      expect(wrapper).toMatchSnapshot();
      // test all text is correct for admin user
      expect(wrapper.find(Popover).at(0).childAt(0).text()).toBe("Account");
      // "Profile" should be in all navbar instances - ie admin/not admin.
      expect(wrapper.find(Menu).at(0).childAt(0).prop("text")).toBe("Profile");
      // expect "Products" to show - which is an admin only option.
      expect(wrapper.find(Menu).at(0).childAt(1).prop("text")).toBe("Products");
      // expect "Create Product" to show - which is an admin only option.
      expect(wrapper.find(Menu).at(0).childAt(2).prop("text")).toBe("Create Product");
      // "Logout" should be in all navbar instances - ie admin/not admin.
      expect(wrapper.find(Menu).at(0).childAt(4).prop("text")).toBe("Logout");
      // check basket is empty
      expect(wrapper.find(Popover).at(1).childAt(0).text()).toBe("Basket (0)");
      expect(wrapper.find(".nav__basket").at(0).childAt(1).text()).toBe(
        "Basket is empty.",
      );
    });

    it("should render correctly when user is NOT admin", () => {
      const wrapper = shallow(
        <NavBar
          signOut={logout}
          setAccountsTab={setAccounts}
          // @ts-ignore
          user={{
            username: "test",
          }}
          userAttributes={{
            sub: "123",
          }}
          items={[]}
          cost={0}
        />,
      );
      expect(wrapper).toMatchSnapshot();
      // find account tab
      expect(wrapper.find(Popover).at(0).childAt(0).text()).toBe("Account");
      // "Profile" should be in all navbar instances - ie admin/not admin.
      expect(wrapper.find(Menu).at(0).childAt(0).prop("text")).toBe("Profile");
      // expect "Orders" to show, which is a non-admin only property
      expect(wrapper.find(Menu).at(0).childAt(1).prop("text")).toBe("Orders");
      // "Logout" should be in all navbar instances - ie admin/not admin.
      expect(wrapper.find(Menu).at(0).childAt(3).prop("text")).toBe("Logout");
      // check basket is empty
      expect(wrapper.find(Popover).at(1).childAt(0).text()).toBe("Basket (0)");
      expect(wrapper.find(".nav__basket").at(0).childAt(1).text()).toBe(
        "Basket is empty.",
      );
    });

    it("should open popover when clicking the account nav link, then close when clicking again", () => {
      const wrapper = shallow(
        <NavBar
          signOut={logout}
          setAccountsTab={setAccounts}
          // @ts-ignore
          user={{ username: "test" }}
          userAttributes={{ sub: "123" }}
          admin
          items={[]}
        />,
      );
      // open accounts popover
      wrapper.find(Popover).at(0).childAt(0).simulate("click");
      // should set menu open to true
      expect(wrapper.state("menuOpen")).toBeTruthy();
      // click the same popover
      wrapper.find(Popover).at(0).childAt(0).simulate("click");
      // should close menu with menuOpen being false
      expect(wrapper.state("menuOpen")).toBeFalsy();
    });

    it("should call the logout function when clicking the button", () => {
      const wrapper = shallow(
        <NavBar
          signOut={logout}
          setAccountsTab={setAccounts}
          // @ts-ignore
          user={{ username: "test" }}
          userAttributes={{ sub: "123" }}
          admin
          items={[]}
        />,
      );
      wrapper.find(".nav__logout").at(0).simulate("click");
      expect(logout).toHaveBeenCalledTimes(1);
    });

    it("should show items in basket NavLink text when they are present in props, and show up in the popover", () => {
      const wrapper = shallow(
        <NavBar
          signOut={logout}
          setAccountsTab={setAccounts}
          // @ts-ignore
          user={{ username: "test" }}
          userAttributes={{ sub: "123" }}
          admin
          items={[]}
          cost={0}
        />,
      );
      // check basket is empty and shows correct text inside popover
      expect(wrapper.find(Popover).at(1).childAt(0).text()).toBe("Basket (0)");
      expect(wrapper.find(".nav__basket").at(0).childAt(1).text()).toBe(
        "Basket is empty.",
      );
      // add item to props
      wrapper.setProps({
        items: [firstItem],
      });
      // popover root should show there is an item in the basket
      expect(wrapper.find(Popover).at(1).childAt(0).text()).toBe("Basket (1)");
      // nav__basket should have the item in the basket popover.
      expect(wrapper.find(".nav__basket").at(0).childAt(1).childAt(0).text()).toBe("1.");
      expect(wrapper.find(".nav__basket-details").at(0).text()).toBe(
        "first - £6.00 + £4.00",
      );
      // add another item to items
      wrapper.setProps({
        items: [firstItem, secondItem],
      });
      // popover content should show this is added to basket with details.
      expect(wrapper.find(Popover).at(1).childAt(0).text()).toBe("Basket (2)");
      expect(wrapper.find(".nav__basket-details").at(1).text()).toBe(
        "second - £6.00 + £4.00",
      );
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("test suite for actions, snapshots and state changes when dispatching actions in CONNECTED component", () => {
    let wrapper;
    // mock store to test actions are created/dispatched.
    const mockStore = configureStore();
    let store;
    // before each test, reset store to initial state and clear actions.
    beforeEach(() => {
      store = mockStore(initialState);
      wrapper = shallow(
        <ConnectedNavBar
          signOut={logout}
          setAccountsTab={setAccounts}
          // @ts-ignore
          user={{ username: "test" }}
          userAttributes={{ sub: "123" }}
          admin
          store={store}
        />,
      );
    });

    it("should render the connected NavBar component properly", () => {
      expect(wrapper.find(NavBar).length).toBe(1);
      expect(wrapper).toMatchSnapshot();
    });

    it("should dispatch actions to mock store and update state when ADDING to basket", () => {
      expect(wrapper).toMatchSnapshot();
      // add first item to store
      store.dispatch(actions.addToBasket(firstItem));
      // check action is in store
      expect(store.getActions()[0].type).toBe(types.ADD_ITEM_TO_BASKET);
      const expectedState = {
        items: [...initialState.basket.items, firstItem],
        cost: 10,
      };
      // test reducer sets store to correct state
      expect(
        reducer(initialState.basket, { type: types.ADD_ITEM_TO_BASKET, item: firstItem }),
      ).toEqual(expectedState);
      // add another item
      store.dispatch(actions.addToBasket(secondItem));
      // check action is in store
      expect(store.getActions()[1].type).toBe(types.ADD_ITEM_TO_BASKET);
      // check reducer updates state with the second item
      expect(
        reducer(expectedState, { type: types.ADD_ITEM_TO_BASKET, item: secondItem }),
      ).toEqual({
        items: [...expectedState.items, secondItem],
        cost: 20,
      });
    });

    it("should dispatch actions to mock store and update state when REMOVING from basket", () => {
      const initialState = {
        items: [firstItem, secondItem],
        cost: 20,
      };
      // remove first item (id is 123)
      store.dispatch(actions.removeFromBasket("123"));
      // expect action to be dispatched in store
      expect(store.getActions()[0].type).toBe(types.REMOVE_ITEM_FROM_BASKET);
      // expect reducer to show state with first item removed
      expect(
        reducer(initialState, { type: types.REMOVE_ITEM_FROM_BASKET, itemID: "123" }),
      ).toEqual({
        items: [secondItem],
        cost: 10,
      });
      // test with one item in store
      const updatedState = {
        items: [secondItem],
        cost: 10,
      };
      store.dispatch(actions.removeFromBasket("321"));
      // should return empty store with cost updated
      expect(
        reducer(updatedState, { type: types.REMOVE_ITEM_FROM_BASKET, itemID: "321" }),
      ).toEqual({
        items: [],
        cost: 0,
      });
      // test where there are no items in array and still trying to remove => shouldn't fail
      expect(
        reducer(
          { items: [], cost: 0 },
          { type: types.REMOVE_ITEM_FROM_BASKET, itemID: "123" },
        ),
      ).toEqual({ items: [], cost: 0 });
    });

    it("should dispatch actions to mock store and update when CLEARING basket", () => {
      // create variable which the store will be in initially
      const initialState = {
        items: [firstItem, secondItem],
        cost: 20,
      };
      // dispatch the clearBasket action
      store.dispatch(actions.clearBasket());
      // expect the first item actions to be of the same type
      expect(store.getActions()[0].type).toBe(types.CLEAR_BASKET);
      // test reducer to see if it's cleared all items and reset cost
      expect(reducer(initialState, { type: types.CLEAR_BASKET })).toEqual({
        items: [],
        cost: 0,
      });
      // same test with one item in store
      expect(
        reducer({ items: [firstItem], cost: 0 }, { type: types.CLEAR_BASKET }),
      ).toEqual({
        items: [],
        cost: 0,
      });
      // should still function when action is dispatched but theres an empty store
      expect(reducer({ items: [], cost: 0 }, { type: types.CLEAR_BASKET })).toEqual({
        items: [],
        cost: 0,
      });
    });
  });
});
