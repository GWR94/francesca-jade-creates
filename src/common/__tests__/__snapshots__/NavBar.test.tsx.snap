// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> test cases for UNCONNECTED component's rendering and snapshots should render correctly when user IS admin 1`] = `
<Fragment>
  <Headroom
    calcHeightOnResize={true}
    disable={false}
    disableInlineStyles={false}
    downTolerance={0}
    onPin={[Function]}
    onUnfix={[Function]}
    onUnpin={[Function]}
    parent={[Function]}
    pinStart={0}
    upTolerance={5}
    wrapperStyle={
      Object {
        "height": "50px",
        "position": "relative",
        "zIndex": 3,
      }
    }
  >
    <Navbar
      className="nav__bar animated slideInDown"
      expand="md"
      light={true}
      tag="nav"
    >
      <img
        alt="Francesca Jade Creates"
        className="navbar__logo"
        src=""
      />
      <NavbarToggler
        onClick={[Function]}
        tag="button"
        type="button"
      />
      <Collapse
        appear={false}
        enter={true}
        exit={true}
        in={false}
        isOpen={false}
        mountOnEnter={false}
        navbar={true}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        tag="div"
        timeout={350}
        unmountOnExit={false}
      >
        <Nav
          className="mr-auto"
          navbar={true}
          tag="ul"
          vertical={false}
        >
          <NavLink
            activeClassName="nav__link--active"
            className="nav__link"
            exact={true}
            onClick={[Function]}
            to="/"
          >
            <i
              className="fas fa-home nav__icon"
            />
            Home
          </NavLink>
          <NavLink
            activeClassName="nav__link--active"
            className="nav__link"
            onClick={[Function]}
            to="/cakes"
          >
            <i
              className="fas fa-birthday-cake nav__icon"
            />
            Cakes
          </NavLink>
          <NavLink
            activeClassName="nav__link--active"
            className="nav__link"
            onClick={[Function]}
            to="/creates"
          >
            <i
              className="fas fa-camera nav__icon"
            />
            Creates
          </NavLink>
        </Nav>
        <Nav
          navbar={true}
          tag="ul"
          vertical={false}
        >
          <Blueprint3.Popover
            boundary="scrollParent"
            captureDismiss={false}
            defaultIsOpen={false}
            disabled={false}
            fill={false}
            hasBackdrop={false}
            hoverCloseDelay={300}
            hoverOpenDelay={150}
            inheritDarkTheme={true}
            interactionKind="click"
            minimal={false}
            modifiers={Object {}}
            openOnTargetFocus={true}
            position="bottom-right"
            targetTagName="span"
            transitionDuration={300}
            usePortal={true}
            wrapperTagName="span"
          >
            <div
              className="nav__link"
              onClick={[Function]}
              role="button"
              tabIndex={0}
            >
              <i
                className="fas fa-user nav__icon"
              />
              Account
            </div>
            <Blueprint3.Menu
              className="nav__menu"
            >
              <Blueprint3.MenuItem
                disabled={false}
                icon={
                  <i
                    className="fas fa-id-badge nav__dropdown-icon"
                  />
                }
                multiline={false}
                onClick={[Function]}
                popoverProps={Object {}}
                shouldDismissPopover={true}
                text="Profile"
              />
              <Blueprint3.MenuItem
                disabled={false}
                icon={
                  <i
                    className="fas fa-shopping-basket nav__dropdown-icon"
                  />
                }
                multiline={false}
                onClick={[Function]}
                popoverProps={Object {}}
                shouldDismissPopover={true}
                text="Products"
              />
              <Blueprint3.MenuItem
                disabled={false}
                icon={
                  <i
                    className="fas fa-plus-square nav__dropdown-icon"
                  />
                }
                multiline={false}
                onClick={[Function]}
                popoverProps={Object {}}
                shouldDismissPopover={true}
                text="Create Product"
              />
              <Blueprint3.MenuDivider />
              <Blueprint3.MenuItem
                className="nav__logout"
                disabled={false}
                icon={
                  <i
                    className="fas fa-sign-out-alt nav__dropdown-icon"
                  />
                }
                multiline={false}
                onClick={[Function]}
                popoverProps={Object {}}
                shouldDismissPopover={true}
                text="Logout"
              />
            </Blueprint3.Menu>
          </Blueprint3.Popover>
          <Blueprint3.Popover
            boundary="scrollParent"
            captureDismiss={false}
            defaultIsOpen={false}
            disabled={false}
            fill={false}
            hasBackdrop={false}
            hoverCloseDelay={300}
            hoverOpenDelay={150}
            inheritDarkTheme={true}
            interactionKind="click"
            minimal={false}
            modifiers={Object {}}
            openOnTargetFocus={true}
            position="bottom"
            targetTagName="span"
            transitionDuration={300}
            usePortal={true}
            wrapperTagName="span"
          >
            <div
              className="nav__link"
              role="button"
              tabIndex={0}
            >
              <i
                className="fas fa-shopping-basket nav__icon"
              />
              Basket (
              0
              )
            </div>
            <div
              className="nav__basket"
            >
              <h5
                className="nav__basket-title"
              >
                Basket
              </h5>
              <p>
                Basket is empty.
              </p>
              <div
                style={
                  Object {
                    "display": "flex",
                    "justifyContent": "space-around",
                    "marginTop": 8,
                  }
                }
              >
                <Blueprint3.Button
                  intent="primary"
                  minimal={true}
                  onClick={[Function]}
                  small={true}
                  text="View Basket"
                />
                <Blueprint3.Button
                  intent="success"
                  small={true}
                  text="Checkout"
                />
              </div>
            </div>
          </Blueprint3.Popover>
        </Nav>
      </Collapse>
    </Navbar>
  </Headroom>
</Fragment>
`;

exports[`<NavBar /> test cases for UNCONNECTED component's rendering and snapshots should render correctly when user is NOT admin 1`] = `
<Fragment>
  <Headroom
    calcHeightOnResize={true}
    disable={false}
    disableInlineStyles={false}
    downTolerance={0}
    onPin={[Function]}
    onUnfix={[Function]}
    onUnpin={[Function]}
    parent={[Function]}
    pinStart={0}
    upTolerance={5}
    wrapperStyle={
      Object {
        "height": "50px",
        "position": "relative",
        "zIndex": 3,
      }
    }
  >
    <Navbar
      className="nav__bar animated slideInDown"
      expand="md"
      light={true}
      tag="nav"
    >
      <img
        alt="Francesca Jade Creates"
        className="navbar__logo"
        src=""
      />
      <NavbarToggler
        onClick={[Function]}
        tag="button"
        type="button"
      />
      <Collapse
        appear={false}
        enter={true}
        exit={true}
        in={false}
        isOpen={false}
        mountOnEnter={false}
        navbar={true}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        tag="div"
        timeout={350}
        unmountOnExit={false}
      >
        <Nav
          className="mr-auto"
          navbar={true}
          tag="ul"
          vertical={false}
        >
          <NavLink
            activeClassName="nav__link--active"
            className="nav__link"
            exact={true}
            onClick={[Function]}
            to="/"
          >
            <i
              className="fas fa-home nav__icon"
            />
            Home
          </NavLink>
          <NavLink
            activeClassName="nav__link--active"
            className="nav__link"
            onClick={[Function]}
            to="/cakes"
          >
            <i
              className="fas fa-birthday-cake nav__icon"
            />
            Cakes
          </NavLink>
          <NavLink
            activeClassName="nav__link--active"
            className="nav__link"
            onClick={[Function]}
            to="/creates"
          >
            <i
              className="fas fa-camera nav__icon"
            />
            Creates
          </NavLink>
        </Nav>
        <Nav
          navbar={true}
          tag="ul"
          vertical={false}
        >
          <Blueprint3.Popover
            boundary="scrollParent"
            captureDismiss={false}
            defaultIsOpen={false}
            disabled={false}
            fill={false}
            hasBackdrop={false}
            hoverCloseDelay={300}
            hoverOpenDelay={150}
            inheritDarkTheme={true}
            interactionKind="click"
            minimal={false}
            modifiers={Object {}}
            openOnTargetFocus={true}
            position="bottom-right"
            targetTagName="span"
            transitionDuration={300}
            usePortal={true}
            wrapperTagName="span"
          >
            <div
              className="nav__link"
              onClick={[Function]}
              role="button"
              tabIndex={0}
            >
              <i
                className="fas fa-user nav__icon"
              />
              Account
            </div>
            <Blueprint3.Menu
              className="nav__menu"
            >
              <Blueprint3.MenuItem
                disabled={false}
                icon={
                  <i
                    className="fas fa-id-badge nav__dropdown-icon"
                  />
                }
                multiline={false}
                onClick={[Function]}
                popoverProps={Object {}}
                shouldDismissPopover={true}
                text="Profile"
              />
              <Blueprint3.MenuItem
                disabled={false}
                icon={
                  <i
                    className="fas fa-envelope-open nav__dropdown-icon"
                  />
                }
                multiline={false}
                onClick={[Function]}
                popoverProps={Object {}}
                shouldDismissPopover={true}
                text="Orders"
              />
              <Blueprint3.MenuDivider />
              <Blueprint3.MenuItem
                className="nav__logout"
                disabled={false}
                icon={
                  <i
                    className="fas fa-sign-out-alt nav__dropdown-icon"
                  />
                }
                multiline={false}
                onClick={[Function]}
                popoverProps={Object {}}
                shouldDismissPopover={true}
                text="Logout"
              />
            </Blueprint3.Menu>
          </Blueprint3.Popover>
          <Blueprint3.Popover
            boundary="scrollParent"
            captureDismiss={false}
            defaultIsOpen={false}
            disabled={false}
            fill={false}
            hasBackdrop={false}
            hoverCloseDelay={300}
            hoverOpenDelay={150}
            inheritDarkTheme={true}
            interactionKind="click"
            minimal={false}
            modifiers={Object {}}
            openOnTargetFocus={true}
            position="bottom"
            targetTagName="span"
            transitionDuration={300}
            usePortal={true}
            wrapperTagName="span"
          >
            <div
              className="nav__link"
              role="button"
              tabIndex={0}
            >
              <i
                className="fas fa-shopping-basket nav__icon"
              />
              Basket (
              0
              )
            </div>
            <div
              className="nav__basket"
            >
              <h5
                className="nav__basket-title"
              >
                Basket
              </h5>
              <p>
                Basket is empty.
              </p>
              <div
                style={
                  Object {
                    "display": "flex",
                    "justifyContent": "space-around",
                    "marginTop": 8,
                  }
                }
              >
                <Blueprint3.Button
                  intent="primary"
                  minimal={true}
                  onClick={[Function]}
                  small={true}
                  text="View Basket"
                />
                <Blueprint3.Button
                  intent="success"
                  small={true}
                  text="Checkout"
                />
              </div>
            </div>
          </Blueprint3.Popover>
        </Nav>
      </Collapse>
    </Navbar>
  </Headroom>
</Fragment>
`;

exports[`<NavBar /> test cases for UNCONNECTED component's rendering and snapshots should show items in basket NavLink text when they are present in props, and show up in the popover 1`] = `
<Fragment>
  <Headroom
    calcHeightOnResize={true}
    disable={false}
    disableInlineStyles={false}
    downTolerance={0}
    onPin={[Function]}
    onUnfix={[Function]}
    onUnpin={[Function]}
    parent={[Function]}
    pinStart={0}
    upTolerance={5}
    wrapperStyle={
      Object {
        "height": "50px",
        "position": "relative",
        "zIndex": 3,
      }
    }
  >
    <Navbar
      className="nav__bar animated slideInDown"
      expand="md"
      light={true}
      tag="nav"
    >
      <img
        alt="Francesca Jade Creates"
        className="navbar__logo"
        src=""
      />
      <NavbarToggler
        onClick={[Function]}
        tag="button"
        type="button"
      />
      <Collapse
        appear={false}
        enter={true}
        exit={true}
        in={false}
        isOpen={false}
        mountOnEnter={false}
        navbar={true}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        tag="div"
        timeout={350}
        unmountOnExit={false}
      >
        <Nav
          className="mr-auto"
          navbar={true}
          tag="ul"
          vertical={false}
        >
          <NavLink
            activeClassName="nav__link--active"
            className="nav__link"
            exact={true}
            onClick={[Function]}
            to="/"
          >
            <i
              className="fas fa-home nav__icon"
            />
            Home
          </NavLink>
          <NavLink
            activeClassName="nav__link--active"
            className="nav__link"
            onClick={[Function]}
            to="/cakes"
          >
            <i
              className="fas fa-birthday-cake nav__icon"
            />
            Cakes
          </NavLink>
          <NavLink
            activeClassName="nav__link--active"
            className="nav__link"
            onClick={[Function]}
            to="/creates"
          >
            <i
              className="fas fa-camera nav__icon"
            />
            Creates
          </NavLink>
        </Nav>
        <Nav
          navbar={true}
          tag="ul"
          vertical={false}
        >
          <Blueprint3.Popover
            boundary="scrollParent"
            captureDismiss={false}
            defaultIsOpen={false}
            disabled={false}
            fill={false}
            hasBackdrop={false}
            hoverCloseDelay={300}
            hoverOpenDelay={150}
            inheritDarkTheme={true}
            interactionKind="click"
            minimal={false}
            modifiers={Object {}}
            openOnTargetFocus={true}
            position="bottom-right"
            targetTagName="span"
            transitionDuration={300}
            usePortal={true}
            wrapperTagName="span"
          >
            <div
              className="nav__link"
              onClick={[Function]}
              role="button"
              tabIndex={0}
            >
              <i
                className="fas fa-user nav__icon"
              />
              Account
            </div>
            <Blueprint3.Menu
              className="nav__menu"
            >
              <Blueprint3.MenuItem
                disabled={false}
                icon={
                  <i
                    className="fas fa-id-badge nav__dropdown-icon"
                  />
                }
                multiline={false}
                onClick={[Function]}
                popoverProps={Object {}}
                shouldDismissPopover={true}
                text="Profile"
              />
              <Blueprint3.MenuItem
                disabled={false}
                icon={
                  <i
                    className="fas fa-shopping-basket nav__dropdown-icon"
                  />
                }
                multiline={false}
                onClick={[Function]}
                popoverProps={Object {}}
                shouldDismissPopover={true}
                text="Products"
              />
              <Blueprint3.MenuItem
                disabled={false}
                icon={
                  <i
                    className="fas fa-plus-square nav__dropdown-icon"
                  />
                }
                multiline={false}
                onClick={[Function]}
                popoverProps={Object {}}
                shouldDismissPopover={true}
                text="Create Product"
              />
              <Blueprint3.MenuDivider />
              <Blueprint3.MenuItem
                className="nav__logout"
                disabled={false}
                icon={
                  <i
                    className="fas fa-sign-out-alt nav__dropdown-icon"
                  />
                }
                multiline={false}
                onClick={[Function]}
                popoverProps={Object {}}
                shouldDismissPopover={true}
                text="Logout"
              />
            </Blueprint3.Menu>
          </Blueprint3.Popover>
          <Blueprint3.Popover
            boundary="scrollParent"
            captureDismiss={false}
            defaultIsOpen={false}
            disabled={false}
            fill={false}
            hasBackdrop={false}
            hoverCloseDelay={300}
            hoverOpenDelay={150}
            inheritDarkTheme={true}
            interactionKind="click"
            minimal={false}
            modifiers={Object {}}
            openOnTargetFocus={true}
            position="bottom"
            targetTagName="span"
            transitionDuration={300}
            usePortal={true}
            wrapperTagName="span"
          >
            <div
              className="nav__link"
              role="button"
              tabIndex={0}
            >
              <i
                className="fas fa-shopping-basket nav__icon"
              />
              Basket (
              2
              )
            </div>
            <div
              className="nav__basket"
            >
              <h5
                className="nav__basket-title"
              >
                Basket
              </h5>
              <div
                className="nav__basket-item"
                key="0"
              >
                <p
                  className="nav__basket-number"
                >
                  1
                  .
                </p>
                <p
                  className="nav__basket-details"
                >
                  first
                   - £
                  6.00
                   + £
                  4.00
                </p>
                <i
                  className="fas fa-times nav__basket-delete"
                  onClick={[Function]}
                  role="button"
                  tabIndex={0}
                />
              </div>
              <div
                className="nav__basket-item"
                key="1"
              >
                <p
                  className="nav__basket-number"
                >
                  2
                  .
                </p>
                <p
                  className="nav__basket-details"
                >
                  second
                   - £
                  6.00
                   + £
                  4.00
                </p>
                <i
                  className="fas fa-times nav__basket-delete"
                  onClick={[Function]}
                  role="button"
                  tabIndex={0}
                />
              </div>
              <div
                style={
                  Object {
                    "display": "flex",
                    "justifyContent": "space-around",
                    "marginTop": 8,
                  }
                }
              >
                <Blueprint3.Button
                  intent="primary"
                  minimal={true}
                  onClick={[Function]}
                  small={true}
                  text="View Basket"
                />
                <Blueprint3.Button
                  intent="success"
                  small={true}
                  text="Checkout"
                />
              </div>
            </div>
          </Blueprint3.Popover>
        </Nav>
      </Collapse>
    </Navbar>
  </Headroom>
</Fragment>
`;

exports[`<NavBar /> test suite for actions, snapshots and state changes when dispatching actions in CONNECTED component should dispatch actions to mock store and update state when ADDING to basket 1`] = `
<ContextProvider
  value={null}
>
  <NavBar
    admin={true}
    items={Array []}
    removeFromBasket={[Function]}
    setAccountsTab={[MockFunction]}
    signOut={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    user={
      Object {
        "username": "test",
      }
    }
    userAttributes={
      Object {
        "sub": "123",
      }
    }
  />
</ContextProvider>
`;

exports[`<NavBar /> test suite for actions, snapshots and state changes when dispatching actions in CONNECTED component should render the connected NavBar component properly 1`] = `
<ContextProvider
  value={null}
>
  <NavBar
    admin={true}
    items={Array []}
    removeFromBasket={[Function]}
    setAccountsTab={[MockFunction]}
    signOut={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    user={
      Object {
        "username": "test",
      }
    }
    userAttributes={
      Object {
        "sub": "123",
      }
    }
  />
</ContextProvider>
`;
